api/src/main/scala/dx/api/DxApi.scala

```diff
@@ -496,6 +496,11 @@ case class DxApi(version: String = "1.0.0", dxEnv: DXEnvironment = DXEnvironment
       case other =>
         throw new Exception(s"unexpected whoami result ${other}")
     }
+    if (id) {
```

Comment:
I don't think this is needed. DXAPI.systemWhoami is 
```
   public static <T> T systemWhoami(Object inputObject, Class<T> outputClass, DXEnvironment env) {
        JsonNode input = mapper.valueToTree(inputObject);
        return DXJSON.safeTreeToValue(
                new DXHTTPRequest(env).request("/system/whoami", input, RetryStrategy.SAFE_TO_RETRY),
                outputClass);
    }
```

Reply:
Which means it's already consistent with the underlying API call dxpy uses 
http://autodoc.dnanexus.com/bindings/python/current/dxpy_api.html?highlight=whoami#dxpy.api.system_whoami
https://documentation.dnanexus.com/developer/api/system-methods#api-method-system-whoami

Reply:
You can delete your folders and re-run tests to confirm this, but I am now seeing folder names builds/user-rnathans and unit_tests/user-rnathans created for me.

Reply:
With this update, the clean_build script removes user-xxxx folders. The removed lines, and probably this API call, would generate user-user-xxxx. 
https://github.com/dnanexus/dxCompiler/pull/113/files#diff-1469ceedc853f6160d183753eb11a262a8cab0848c9c1d1f355b3a5f63750646L14

Reply:
ok thanks